src/coreclr/jit/target.h

```diff
@@ -260,26 +254,41 @@ struct regMaskTP
     void RemoveRegNumFromMask(regNumber reg, var_types type);
     bool IsRegNumInMask(regNumber reg, var_types type) const;
 #endif
-    void             AddGprRegs(SingleTypeRegSet gprRegs);
-    void             AddRegNum(regNumber reg, var_types type);
-    void             AddRegNumInMask(regNumber reg);
-    void             AddRegsetForType(SingleTypeRegSet regsToAdd, var_types type);
-    SingleTypeRegSet GetRegSetForType(var_types type) const;
-    bool             IsRegNumInMask(regNumber reg) const;
-    bool             IsRegNumPresent(regNumber reg, var_types type) const;
-    void             RemoveRegNum(regNumber reg, var_types type);
-    void             RemoveRegNumFromMask(regNumber reg);
-    void             RemoveRegsetForType(SingleTypeRegSet regsToRemove, var_types type);
-
-    regMaskTP(regMaskSmall lowMask, regMaskSmall highMask)
+    void                       AddGprRegs(SingleTypeRegSet gprRegs);
+    void                       AddRegNum(regNumber reg, var_types type);
+    void                       AddRegNumInMask(regNumber reg);
+    void                       AddRegsetForType(SingleTypeRegSet regsToAdd, var_types type);
+    SingleTypeRegSet           GetRegSetForType(var_types type) const;
+    bool                       IsRegNumInMask(regNumber reg) const;
+    bool                       IsRegNumPresent(regNumber reg, var_types type) const;
+    void                       RemoveRegNum(regNumber reg, var_types type);
+    void                       RemoveRegNumFromMask(regNumber reg);
+    void                       RemoveRegsetForType(SingleTypeRegSet regsToRemove, var_types type);
+    static constexpr regMaskTP CreateFromRegNum(regNumber reg, regMaskSmall mask)
+    {
```

Comment:
Is `mask` needed as a parameter? Can't we compute it here as `1 << reg`?

Reply:
predicate registers starts from 64 thru 80 and the mask of them start with `0x0`. I can make it work, but just want to rely on the `mask` that we already have in register files and how we use them currently.

Reply:
For predicate registers it would just be `(1 << (reg - 64))` under the if, I suppose.

Reply:
Sure, I don't have much preference on which way or the other, but will keep this in mind during follow up.